lib/portage/package/ebuild/fetch.py

```diff
@@ -878,7 +886,7 @@ def onerror(e):
 				eout.quiet = mysettings.get("PORTAGE_QUIET") == "1"
 				match, mystat = _check_distfile(
 					myfile_path, pruned_digests, eout, hash_filter=hash_filter)
-				if match:
+				if match and not (force and not orig_digests):
```

Comment:
Wouldn't be it more logical for `force` to force even when digests are present? Or am I reading this condition wrong?

Reply:
You are correct, it would be more logical in a sense. However, if there are digests that don't match then it's going to fetch anyway.

Reply:
Could you add an explanatory comment there? Also avoiding the nested negation could be better for the reader. If my memory serves me right, `and (not force or orig_digests)` should be equivalent.

Reply:
The current logic allows `force=True` to be specified without forcing an unnecessary fetch when the digests match.

Reply:
That sounds illogical. Maybe it'd be better to error out if `force` is passed and digests are present at the same time.

Reply:
Ok, I've removed the orig_digests logic and added this condition at the beginning of the function:
```python
if force and digests:
	# Since the force parameter can trigger unnecessary fetch when the
	# digests match, do not allow force=True when digests are provided.
	raise PortageException(_('fetch: force=True is not allowed when digests are provided'))
```